scripts/lib/CIME/namelist.py

```diff
@@ -2047,12 +2098,15 @@ def _parse_namelist_group(self):
             self._settings[group_name] = {}
         self._eat_whitespace()
         while self._curr() != '/':
-            name, values = self._parse_name_and_values()
+            name, values, addto = self._parse_name_and_values()
             dsettings = []
             if self._groupless:
                 if name in self._settings:
                     dsettings = self._settings[name]
-                values = merge_literal_lists(dsettings, values)
+                    if addto:
+                        values = self._settings[name] + values
+                if not addto:
```

Comment:
This looks like it works correctly now, but I'm still wondering why it looks different from the analogous code below:

```python
                if name in self._settings:
                    if addto:
                        values = self._settings[name] + values
                    else:
                        values = merge_literal_lists(self._settings[name], values)
```

My understanding is that the two blocks of code are doing the same thing, so I'd find it easier to read & understand this code if they looked identical (or called some shared function). I realize that they looked a little different in the baseline code, but now that this code block is getting more complicated, it would be good to make them uniform, so that the true differences between the two don't get lost in accidental differences like this.

This would amount to a small logic change, in that the 'if not addto' block would now only be executed if name in self._settings is true; but if I understand the code correctly, that shouldn't matter, because: if (not addto) is true but (name in self._settings) is false, then you're calling merge_literal_lists with dsettings = [], which looks to me like a no-op.